dockstore-client/src/main/java/io/dockstore/client/cli/CheckerClient.java

```diff
@@ -393,15 +393,26 @@ private Entry getEntryFromPath(String entryPath) {
      * @param entry
      * @return checker workflow
      */
-    private Workflow getCheckerWorkflowFromEntry(Entry entry) {
+    private Workflow getCheckerWorkflowFromEntry(Entry entry, boolean authRequired) {
```

Comment:
I think this could reuse/call getDockstoreWorkflow twice if it becomes a protected method https://github.com/ga4gh/dockstore/blob/develop/dockstore-client/src/main/java/io/dockstore/client/cli/nested/WorkflowClient.java#L295 
(i.e. get the workflow using getDockstoreWorkflow, get the checker workflow id, and then call getDockstoreWorkflow again)  

I think this is a pattern that will come up several times and I'd like to be able to re-use the code/optimize it as a group later

Reply:
getDockstoreWorkflow uses the path as input, however in this case it is more helpful to use the id. So I'll make two functions, getDockstoreWorkflowById and getDockstoreWorkflowByPath.